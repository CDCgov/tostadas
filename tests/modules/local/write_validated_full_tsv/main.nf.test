nextflow_process {

    name "Test Process WRITE_VALIDATED_FULL_TSV"
    script "modules/local/write_validated_full_tsv/main.nf"
    process "WRITE_VALIDATED_FULL_TSV"

    test("Should concatenate TSVs and keep only the first header") {

        when {
            params {

            }

            process {
                """
                input[0] = [
                    file("${projectDir}/tests/modules/local/write_validated_full_tsv/batch_1.tsv")
                ]
                """
            }
        }

        then {
            assert process.success
            assert file("${process.out.validated_concatenated_tsv[0]}").exists()
            assert snapshot(file("${process.out.validated_concatenated_tsv[0]}")).match()
        }
    }
}
