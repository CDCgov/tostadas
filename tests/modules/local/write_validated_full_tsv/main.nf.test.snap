{
    "Should concatenate TSVs and keep only the first header": {
        "content": [
            "validated_metadata_all_samples.tsv:md5,e3235a94deedd67e93edc18d0804422b"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:23:19.044621161"
    }
}